src/server/src/Server.cpp

```diff
@@ -253,37 +291,41 @@ Server::State Server::get_state()
     return state;
 }
 
-
 void Server::stop_motion()
 {
     drdStop();
 }
 
-
-void Server::set_state(const State& state)
+void Server::set_state(const State &state)
 {
     mutex_.lock();
     state_ = state;
     mutex_.unlock();
 }
 
-
 void Server::stream_robot_state()
 {
-    double position[3];
+    double position[3]; // position px, py, pz
 
     dhdGetPosition(&position[0], &position[1], &position[2]);
 
-    Vector& position_out = port_position_.prepare();
+    Vector &position_out = port_position_.prepare();
     position_out = Vector(3, position);
     port_position_.write();
 
+    double velocity[3]; // velocity vx, vy, vz
```

Comment:
See the comment above about the possibility to send positions and velocities together in the same port such that they are synchronized.

Reply:
what do you mean by that? create something like double robot_state[6] and then assign position to the first 3 and velocity to the last 3?

Reply:
Yes :) that exactly what I meant :)

Reply:
Now all robot states (position, velocity and force) are together and with that synchronized

Reply:
Ok, thank you!